### ISSUE TITLE
Bug running cronjobs with set timezones using node-cron

### ISSUE THREAD

**Issue by sum117** [oven-sh/bun#5385]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

If the timezone property is set, nothing will be ran.
```js
import cron from 'node-cron';
cron.schedule('* * * * *', () => {
    console.log("Cron 1 executed in:", new Date().toLocaleString());
}, {
    timezone: "America/Sao_Paulo"
})
```

This works normally in Node.JS though. Thankfully, I've found a workaround:

```js
import cron from 'node-cron';

Bun.env.TZ = "America/Sao_Paulo";

cron.schedule('* * * * *', () => {
    console.log("Cron 1 executed in:", new Date().toLocaleString());
})

```

Now that works normally, but I don't think this is how it should happen. Please let me know what you think of this bug.

### What is the expected behavior?

Bun reacts normally to the timezone property from the `schedule()` function in `node-cron` module.

### What do you see instead?

The execution doesn't happen at all.

**Comment by Electroid** [comment]
I wonder if this is an issue with our `Date` parser or `Intl.DateTimeFormat`. I found this code in `node-cron`:

```js
const dtf = new Intl.DateTimeFormat('en-US', {
  year: 'numeric',
  month: '2-digit',
  day: '2-digit',
  hour: '2-digit',
  minute: '2-digit',
  second: '2-digit',
  hourCycle: 'h23',
  fractionalSecondDigits: 3,
  timeZone: this.timezone
});
return new Date(dtf.format(date));
```

**Comment by sum117** [comment]
Issue seems to be fixed in v1.0.2
